CodeEdit/Features/AppPreferences/Sections/TextEditingPreferences/TextEditingPreferencesView.swift

```diff
@@ -130,6 +133,72 @@ struct TextEditingPreferencesView: View {
         }
     }
 
+    private var textEncoding: some View {
+        Picker("Default Text Encoding", selection: $prefs.preferences.textEditing.textEncoding) {
+            Group {
+                Text("ASCII")
+                    .tag(String.Encoding.ascii)
+                Text("Non Lossy ASCII")
+                    .tag(String.Encoding.nonLossyASCII)
+
+                Divider()
+
+                Text("ISO 2022 Japan")
+                    .tag(String.Encoding.iso2022JP)
+                Text("Japanese EUC")
+                    .tag(String.Encoding.japaneseEUC)
+
+                Divider()
+
+                Text("ISO Latin 1")
+                    .tag(String.Encoding.isoLatin1)
+                Text("ISO Latin 2")
+                    .tag(String.Encoding.isoLatin2)
+                Text("macOS Roman")
+                    .tag(String.Encoding.macOSRoman)
+            }
+
+            Group {
+                Divider()
+
+                Text("Nextstep")
+                    .tag(String.Encoding.nextstep)
+                Text("Shift JIS")
+                    .tag(String.Encoding.shiftJIS)
+                Text("Symbol")
+                    .tag(String.Encoding.symbol)
+
+                Divider()
+            }
+
+            Group {
+                Text("Unicode")
+                    .tag(String.Encoding.unicode)
+
+                Text("Unicode (UTF-8)")
+                    .tag(String.Encoding.utf8)
+
+                Divider()
+
+                Text("Unicode (UTF-16)")
+                    .tag(String.Encoding.utf16)
+                Text("Unicode (UTF-16) Big Endian")
+                    .tag(String.Encoding.utf16be)
+                Text("Unicode (UTF-16) Little Endian")
+                    .tag(String.Encoding.utf16le)
+
+                Divider()
+
+                Text("Unicode (UTF-32)")
+                    .tag(String.Encoding.utf32)
+                Text("Unicode (UTF-32) Big Endian")
+                    .tag(String.Encoding.utf32be)
+                Text("Unicode (UTF-32) Little Endian")
+                    .tag(String.Encoding.utf32le)
+            }
+        }
+    }
+
```

Comment:
- This still needs to be improved to use loops (e.g. `ForEach(…) {…}` - the current approach is just a lot of unnecessary repetitive code. 

- Also the UTF section should be on top since this is the most common.